src/librustc_session/options.rs

```diff
@@ -968,4 +981,6 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
         "link the `.rlink` file generated by `-Z no-link`"),
     new_llvm_pass_manager: Option<bool> = (None, parse_opt_bool, [TRACKED],
         "use new LLVM pass manager"),
+    src_file_checksum: Option<SourceFileChecksumKind> = (None, parse_src_file_checksum, [UNTRACKED],
+        "embed a cryptographic hash of source files in debug info (`md5`, or `sha1`)"),
```

Comment:
Why not always embed a `SHA-1` checksum?

Reply:
Ultimately, I'd like to add `SHA-256` and make that the default, however, LLVM does not support it yet. I do have a patch to add `SHA-256` to LLVM, but I haven't sent it to them yet.

If we made `SHA-1`, the default now, would we be comfortable changing the default in the future to `SHA-256`?

Reply:
I don't think we provide any stability guarantees regarding the exact DWARF being emitted. I think it only matters that the emitted DWARF makes a program debug-able using gdb, lldb and maybe other debuggers. Changing the hash algorithm shouldn't have any effect on that.

Reply:
Can you put this information in the PR description? Maybe even in the command, e.g.:
```
either `md5` (DWARF/PDB) or `sha1` (PDB-only)
```

The hash we already use (`src_hash`) is (or we truncate it to) 128-bit, so using MD5 wouldn't be much worse (except for it being vulnerable to certain attacks, which I don't know how much we care about).